airbyte-webapp/src/views/Connector/ConnectorForm/components/Property/SecretConfirmationControl.tsx

```diff
@@ -0,0 +1,114 @@
+import { useField, useFormikContext } from "formik";
+import React, { useEffect, useRef, useState } from "react";
+import { FormattedMessage } from "react-intl";
+import styled from "styled-components";
+
+import { Button } from "components/ui/Button";
+import { Input } from "components/ui/Input";
+import { SecretTextArea } from "components/ui/SecretTextArea";
+
+const ComponentContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`;
+
+interface SecretConfirmationControlProps {
+  showButtons?: boolean;
+  name: string;
+  multiline: boolean;
+  disabled?: boolean;
+  error?: boolean;
+}
+
+const SecretConfirmationControl: React.FC<SecretConfirmationControlProps> = ({
+  showButtons,
+  disabled,
+  multiline,
+  name,
+  error,
+}) => {
+  const [field, , helpers] = useField(name);
+  const [previousValue, setPreviousValue] = useState<unknown>(undefined);
+  const isEditInProgress = Boolean(previousValue);
+  const controlRef = useRef<HTMLElement>(null);
+
+  const { dirty, touched } = useFormikContext();
+
+  const component =
+    multiline && (isEditInProgress || !showButtons) ? (
+      <SecretTextArea
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        rows={3}
+        disabled={disabled}
+        error={error}
+      />
+    ) : (
+      <Input
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        type="password"
+        disabled={disabled}
+        error={error}
+      />
+    );
+
+  useEffect(() => {
+    if (!dirty && !touched && previousValue) {
+      setPreviousValue(undefined);
+    }
+  }, [dirty, helpers, previousValue, touched]);
```

Comment:
> Listen to the global formik state - if the dirty flag flips to false but there is a "previous value", reset that as well

Just to confirm my understanding: this code is basically ensuring that if the formik form is reset back to the initial state, then the `previousValue` state will be cleared out? An in practice, this would happen if a user clicks the `Cancel` button at the bottom of the ConnectorForm

Reply:
Exactly, if the form-level "Cancel" button is hit, all open secret controls are "closed" as well.